ksql-engine/src/main/java/io/confluent/ksql/function/udf/string/MaskKeepLeftKudf.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ **/
+
+package io.confluent.ksql.function.udf.string;
+
+import io.confluent.ksql.function.KsqlFunctionException;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+
+@UdfDescription(name = "mask_keep_left", author = "Confluent",
+    description = "Returns a version of the input string with all but the"
+        + " specified number of left-most characters masked out."
+        + " Default masking rules will replace all upper-case characters with 'X', all lower-case"
+        + " characters with 'x', all digits with 'n', and any other character with '-'.")
+public class MaskKeepLeftKudf {
+
+  @Udf(description = "Returns a masked version of the input string. All characters except for the"
+      + " first n will be replaced according to the default masking rules.")
+  public String mask(final String input, final int numChars) {
+    return doMask(new Masker(), input, numChars);
+  }
+
+  @Udf(description = "Returns a masked version of the input string. All characters except for the"
+      + " first n will be replaced with the specified masking characters: e.g."
+      + " mask_keep_left(input, numberToKeep, upperCaseMask, lowerCaseMask, digitMask, otherMask)"
+      + " . Pass NULL for any of the mask characters to prevent masking of that character type.")
+  public String mask(final String input, final int numChars, final String upper, final String lower,
+      final String digit, final String other) {
+    // TODO once KSQL gains Char sql-datatype support we should change the xxMask params to int
+    // (codepoint) instead of String
+
+    final int upperMask = Masker.getMaskCharacter(upper);
+    final int lowerMask = Masker.getMaskCharacter(lower);
+    final int digitMask = Masker.getMaskCharacter(digit);
+    final int otherMask = Masker.getMaskCharacter(other);
+    final Masker masker = new Masker(upperMask, lowerMask, digitMask, otherMask);
+    return doMask(masker, input, numChars);
+  }
+
+  private String doMask(final Masker masker, final String input, final int numChars) {
+    validateParams(numChars);
```

Comment:
I would make this a utility method which also checks that the input masks are actually single characters, in addition to checking that `numChars` is positive. The errror message can be generic, maybe something like `The masking function only accepts single character masks", or "The number of characters to mask or skip must be positive". 

Further, I would call this function from the `mask` method of all the `Mask`, `MaskKeepLeft`, `MaskKeepRight`, `MaskLeft`, and `MaskRight` classes.

This utility method could even be a static parameter in `Masker` for now.